.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,171 @@
+            fetch-depth: 0
+            fetch-tags: true
+            ref: ${{ env.REF_ID }}
+
+      - id: dev_image_tag
+        name: Set Docker Stag image tag for updates of the DEV branch
+        if: github.ref == 'refs/heads/dev'
+        run: |
+
+          echo TAG ="dev-$(git describe --tags --abbrev=7)" >> "$GITHUB_ENV"
+
+          # Updates to the main branch are deployed to stage.
+          echo DEPLOYMENT_ENV=dev >> "$GITHUB_ENV"
+          echo DEPLOYMENT_REALM=nonprod >> "$GITHUB_ENV"
+
+      - id: stage_image_tag
+        name: Set Docker Stag image tag for updates of the main branch
+        if: github.ref == 'refs/heads/main'
+        run: |
+
+          echo TAG ="$(git describe --tags --abbrev=7)" >> "$GITHUB_ENV"
+
+          # Updates to the main branch are deployed to stage.
+          echo DEPLOYMENT_ENV=stage >> "$GITHUB_ENV"
+          echo DEPLOYMENT_REALM=nonprod >> "$GITHUB_ENV"
+
+      - id: prod_image_tag
+        name: Set Docker image tag to the git tag for tagged builds
+        if: startsWith(github.ref, 'refs/tags/')
+        run: |
+
+          echo TAG ="$(git describe --tags --abbrev=7)" >> "$GITHUB_ENV"
+          # Version tags are deployed to prod.
+          echo DEPLOYMENT_ENV=prod >> "$GITHUB_ENV"
+          echo DEPLOYMENT_REALM=prod >> "$GITHUB_ENV"
+
+      - id: gcp_auth
+        name: GCP authentication
+        uses: google-github-actions/auth@v2
+        with:
+            token_format: access_token
+            service_account:  artifact-writer@${{ env.GCP_PROJECT_ID }}.iam.gserviceaccount.com
+            workload_identity_provider: ${{ secrets.WORKLOAD_IDENTITY_POOL_PROJECT_NUMBER }}
+
+      - id: docker_login
+        name: Log in to the container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.GAR_LOCATION }}-docker.pkg.dev
+          username: oauth2accesstoken
+          password: ${{ steps.gcp_auth.outputs.access_token }}
+
+
+      - id: build_and_push
+        name: Build and push image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          tags: |
+            ${{ env.GAR_LOCATION }}-docker.pkg.dev/${{ env.GCP_PROJECT_ID }}/${{ env.GAR_REPOSITORY }}/${{ env.IMAGE_NAME }}:${{ env.TAG }}
+
+          push: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+  upload-static-assets:
+    name: upload static assets
+    environment: ${{ needs.build.outputs.deployment_env }}
+    runs-on: ubuntu-latest
+    needs:
+      - build
+
+    permissions:
+      contents: read
+      id-token: write
+
+    steps:
+      - id: gcp_auth
+        name: gcp auth
+        uses: google-github-actions/auth@v2
+        with:
+          token_format: access_token
+          service_account: artifact-writer@${{ env.GCP_PROJECT_ID }}.iam.gserviceaccount.com
+          workload_identity_provider: projects/324168772199/locations/global/workloadIdentityPools/github-actions/providers/github-actions
```

Comment:
artifact-writer can't write to static assets bucket

Reply:
change it to the deploy sa